File: app/src/main/java/com/anokmik/databinding/communicator/TwoWayCommunicator.java
package com.anokmik.databinding.communicator;

import android.content.res.Resources;
import android.support.v4.content.res.ResourcesCompat;
import android.util.Log;

import com.anokmik.databinding.R;

public class TwoWayCommunicator {

    private static final String TAG = TwoWayCommunicator.class.getSimpleName();

    private final Resources resources;

    private String editTextValue;

    private boolean checkBoxValue;

    private int checkedButton;

    private int seekBarProgress;

    private int ratingBarValue;

    private int color;

    public TwoWayCommunicator(Resources resources) {
        this.resources = resources;
        this.checkedButton = R.id.radio_button_first;
        this.color = ResourcesCompat.getColor(resources, R.color.material_primary_red, null);
    }

    public String getEditTextValue() {
        return editTextValue;
    }

    public void setEditTextValue(String newEditTextValue) {
        editTextValue = newEditTextValue;
        Log.i(TAG, "setEditTextValue: " + editTextValue);
    }

    public boolean isCheckBoxValue() {
        return checkBoxValue;
    }

    public void setCheckBoxValue(boolean newCheckBoxValue) {
        checkBoxValue = newCheckBoxValue;
        Log.i(TAG, "setCheckBoxValue: " + checkBoxValue);
    }

    public int getCheckedButton() {
        return checkedButton;
    }

    public void setCheckedButton(int newCheckedButton) {
        checkedButton = newCheckedButton;
        Log.i(TAG, "setCheckedButton: " + getResourceName(checkedButton));
    }

    public int getSeekBarProgress() {
        return seekBarProgress;
    }

    public void setSeekBarProgress(int newSeekBarProgress) {
        seekBarProgress = newSeekBarProgress;
        Log.i(TAG, "setSeekBarProgress: " + seekBarProgress);
    }

    public int getRatingBarValue() {
        return ratingBarValue;
    }

    public void setRatingBarValue(int newRatingBarValue) {
        ratingBarValue = newRatingBarValue;
        Log.i(TAG, "setRatingBarValue: " + ratingBarValue);
    }

    public int getColor() {
        return color;
    }

    public void setColor(int newColor) {
        color = newColor;
        Log.i(TAG, "setColor: " + getColorHex());
    }

    private String getResourceName(int resourceId) {
        return resources.getResourceName(resourceId);
    }

    private String getColorHex() {
        return "#" + Integer.toHexString(color).toUpperCase().substring(2);
    }

}

File: app/src/main/kotlin/com/anokmik/databinding/communicator/TwoWayCommunicator.kt
package com.anokmik.databinding.communicator

import android.content.res.Resources
import android.support.v4.content.res.ResourcesCompat
import android.util.Log

import com.anokmik.databinding.R

class TwoWayCommunicator(private val resources: Resources) {

    var editTextValue = ""
        set(newEditTextValue) {
            field = newEditTextValue
            Log.i(TAG, "setEditTextValue: " + field)
        }

    var checkBoxValue = false
        set(newCheckBoxValue) {
            field = newCheckBoxValue
            Log.i(TAG, "setCheckBoxValue: " + field)
        }

    var checkedButton = R.id.radio_button_first
        set(newCheckedButton) {
            field = newCheckedButton
            Log.i(TAG, "setCheckedButton: " + getResourceName(field))
        }

    var seekBarProgress = 0
        set(newSeekBarProgress) {
            field = newSeekBarProgress
            Log.i(TAG, "setSeekBarProgress: " + field)
        }

    var ratingBarValue = 0
        set(newRatingBarValue: Int) {
            field = newRatingBarValue
            Log.i(TAG, "setRatingBarValue: " + field)
        }

    var color = ResourcesCompat.getColor(resources, R.color.material_primary_red, null)
        set(newColor: Int) {
            field = newColor
            Log.i(TAG, "setColor: " + getColorHex(field))
        }

    private fun getResourceName(resourceId: Int): String {
        return resources.getResourceName(resourceId)
    }

    private fun getColorHex(color: Int): String {
        return "#" + Integer.toHexString(color).toUpperCase().substring(2)
    }

    companion object {
        private val TAG = TwoWayCommunicator::class.java.simpleName
    }

}